apps/vaporgui/VizWin.cpp

```diff
@@ -858,3 +860,58 @@ void VizWin::updateManip(bool initialize)
 
     GL_ERR_BREAK();
 }
+
+void VizWin::_updateSliceOriginGlyph()
+{
+    VAPoR::SliceParams *sp = dynamic_cast<VAPoR::SliceParams *>(_getRenderParams());
+    if (sp == nullptr) return;
+
+    double xOrigin = sp->GetValueDouble(SliceParams::XOriginTag, 0.);
```

Comment:
Ditto here. Let's generalize this so that _updateSliceOriginGlyph can be used for other things besides the slice renderer

Reply:
My thought was that the Contour and Slice renderers would share a base class that would both have an XOriginTag, and could both render the glyph using the same logic in VizWin.  I couldn't think of any use cases outside the Slice and Contour renderers.

The "orientable" flag would fix this, but 1) it introduces inconsistency in how the VizWin operates (see comment above) and 2) each RenderParams class would need GetOrientable() functions and initialization logic.  Are there advantages to using the flag over inheriting from a Slice/Contour base class?

I'm open all week if this would be easier to talk about, or just give me the signal and I can implement the flag.

Reply:
Hard coding is rarely a good practice, which is what testing the class type amounts to. The contour and slice renderer do not share a base class, so this isn't an option without refactoring. Refactoring simply because contour and and slice both support arbitrary cutting planes would paint us into a corner. What happens is another renderer comes along that requires an orientable plane, but also shares some property with the Barb renderer. Do we then need multiple inheritance?  

Something like a RenderParams::GetOrientable() that defaults to false (so other renderers can ignore it) seems like the way to go here.